### DISCUSSION TITLE
Is there a way to rotate traces and generations?

### DISCUSSION THREAD

**Discussion by woung717** [langfuse/langfuse#4429] [Support] [answered]
We’ve set up Langfuse to monitor the usage stats of our LLM application and trace bugs. Since our application makes a significant number of completion requests, it generates a large volume of traces and generation logs.

I don’t need data that was created a long time ago, so I’d like to implement a way to rotate traces and generations by date. However, I couldn’t find any rotation settings or functions in the manual. Does this mean Langfuse doesn’t support rotation at this time?

If that’s the case, would it be okay to manually delete old traces and generations in Postgres? Are there any potential side effects I should be aware of if I remove outdated data directly from the database?

**Comment by marcklingen** [comment, marked as answer]
Have a look here: https://github.com/orgs/langfuse/discussions/2268

**Reply by woung717** [reply]
@marcklingen That’s exactly what I was looking for! Thank you!

**Reply by marcklingen** [reply]
perfect, thanks for confirming, happy to help!
